src/client/src/tscProfile.c

```diff
@@ -282,18 +282,8 @@ int tscBuildQueryStreamDesc(void *pMsg, STscObj *pObj) {
 //      }
       pthread_mutex_lock(&pSql->subState.mutex);
       if (pSql->pSubs != NULL && pSql->subState.states != NULL) {
-        for (int32_t i = 0; i < pQdesc->numOfSub; ++i) {
+        for (int32_t i = 0; i < pSql->subState.numOfSub; ++i) {
```

Comment:
The `pSql->subState.numOfSub` is the size of the array `pSubs`, which are to be synchronized by the mutex.